cds_migrator_kit/videos/weblecture_migration/transform/transform_files.py

```diff
@@ -0,0 +1,342 @@
+
+        return highest_quality_composite, other_composites
+    
+    def get_composite(self, all_files):
+        """Get all the composite files"""
+        master_composite, subformats = self.get_highest_and_other_composites(file_list=all_files)
+        
+        # Add the master composite
+        self.file_info_json["master_path"] = str(self.media_data_folder / master_composite)
+        
+        # Add it's subformats 
+        for file_name in subformats:
+            self.file_info_json["subformats"].append(str(self.media_data_folder / file_name))
+            
+        # Frames (generated using composite)
+        frame_folder = self.media_data_folder / "frames"
+        frames_list = self.get_all_files(frame_folder)
+        # Sort them eg:"frame-1.jpg" to "frame-10.jpg"
+        sorted_frames = sorted(frames_list, key=lambda f: int(re.search(r'frame-(\d+)', f).group(1)))
+        # Add frames
+        for frame in sorted_frames:
+            self.file_info_json["frames"].append(str(frame_folder / frame))
+
+
+    def check_media_data_folder(self, files_paths):
+        """Check the file paths and the media_data folder.
+        
+        files_paths: file paths comes from the record.
+        media_data folder will have composite file, vtt files, subformats, frames."""
+        # No path found in the record
+        if not files_paths:
+            raise UnexpectedValue(
+                message="No media file found in the record!",
+                stage="transform",
+                recid=self.recid,
+                value=path,
+                priority="critical",
+            )
+        # Get all files in media_data folder
+        all_files = self.get_all_files(self.media_data_folder)
+        # No file found in media_data folder
+        if not all_files:
+            raise ManualImportRequired(
+                message="No file found in the media_data folder!",
+                stage="transform",
+                recid=self.recid,
+                value=path,
+                priority="critical",
+            )
+            
+        # Check all files_paths exists in media_data folder
+        for path in files_paths:
+            file_name = Path(path).name
+            if file_name not in all_files:
+                raise ManualImportRequired(
+                    message="File not found in the media_data folder!",
+                    stage="transform",
+                    recid=self.recid,
+                    value=path,
+                    priority="critical",
+                )
+                
+        # Use the composite
+        if self.use_composite:
+            # Get the composite files in the folder
+            self.get_composite(all_files=all_files)
+            # File paths comes from the record will be additional file.
+            # If we use the composite, we dont need the subformats of the presenter and presentation.
+            additional_files = [file for file in files_paths 
```

Comment:
[here](https://github.com/CERNDocumentServer/cds-migrator-kit/pull/217/files#diff-15a362e56ee872ff4b49369e3e257bc4a28237f2729a3b95f6e59593c172568cR131) you also compute the additional files? Can we extract the method or can compute this once?